src/mixins/isMobile/index.js

```diff
@@ -20,26 +20,24 @@
  *
  */
 
+import { IsMobileState } from 'Utils/IsMobileState'
+
 export default {
 	data() {
 		return {
-			isMobile: this._isMobile()
+			isMobile: false
 		}
 	},
-	beforeMount() {
-		window.addEventListener('resize', this._onResize)
+	mounted() {
+		IsMobileState.$on('changed', this.onIsMobileChanged)
+		this.isMobile = IsMobileState.isMobile
```

Comment:
This line should not be needed, as the initial state is `false` for both the mixin and the resize listener and the 'changed' event will fire if the state is changed to `true`. Also this depends on the internal structure of the `IsMobileState` component, which should be avoided.

Reply:
This is needed if a component using this mixing is created after another component already used the mixin. In that case, `IsMobileState.isMobile` could be already set to `true` and no further event is fired. So we definitely need this!

Reply:
True! Did not think of that. Do you think it would make sense to watch changes to the [$listeners](https://vuejs.org/v2/api/#vm-listeners).changed property, so we can fire the 'changed' event every time a new listener joins? This would remove the coupling between the mixin and the component. Don't know if that's overkill though or would even introduce other side effects.

Reply:
I'm not sure, I think it's fine as it is. What do you think, @skjnldsv ?

Reply:
I think we should try as much as possible to mimic the current behaviour which works very well (despite the performance issue) :)

Reply:
Ok, this means LGTM?

Reply:
If it works, yes? :p